skops/hub_utils/_hf_hub.py

```diff
@@ -626,7 +627,13 @@ def get_model_output(repo_id: str, data: Any, token: Optional[str] = None) -> An
     Also note that if the model repo is private, the inference API would not be
     available.
     """
-    model_info = HfApi().model_info(repo_id=repo_id, token=token)
+    # TODO: remove once we drop support for Hub<0.10
+    if Version(huggingface_hub.__version__) <= Version("0.9.1"):
+        extra_kwargs = {"token": token}
+    else:
+        extra_kwargs = {"use_auth_token": token}
```

Comment:
rather have this in `fixes.py` as well.

Reply:
and we don't need to check for version, we try with `use_auth_token`, if it fails we do `token`.

Reply:
Hmm, I understand that try...except allows us to skirt having version checking code. But having a bunch of

```python
try:
    client.foo(repo_id=repo_id, repo_type="model", token=token)
except TypeError:
    client.foo(repo_id=repo_id, repo_type="model", use_auth_token=token)
```

looks suboptimal to me. It litters our code in multiple places, instead of dealing with the issue once. And, just as a stupid example, what if Hub re-introduces the `token` argument but with a different functionality, it could lead to unwanted side-effects.

Reply:
We can have a `Client` class in `fixes.py` which does this, and import that instead, and that class exposes the methods we need in `skops`, and handles different versions there. The rest of the codebase would stay clean this way.

Reply:
That might be a solution, though a bit of a bigger undertaking with some open questions:

* private or public?
* automatically expose all methods or only those we use?
* handle versions with try...expect or version checks?
* also handle inference api?
* testing will probably not be quite trivial

As a quick fix for this issue here, we could also pin the `huggingface_hub` version and close this PR. WDYT?

Reply:
Since we don't have many users yet, I'm also okay with closing this, and then moving along with using modelcards from the new release, which would mean depending on the latest one anyway, and we won't have this issue. So happy to close (I'd leave that to you).